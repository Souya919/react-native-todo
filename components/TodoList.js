import React from "react";
import { View, TouchableOpacity, Text } from "react-native";
import { Entypo, MaterialIcons } from "@expo/vector-icons";

export default function TodoList({ item, editItem, deleteItem }) {
  return (
    <View style={styles.componentContainer}>
      <TouchableOpacity style={styles.listContainer}>
        <View style={styles.circleContainer}>
          <Entypo name="circle" size={20} color="midnightblue" />
        </View>
        <View>
          <Text style={styles.textItem}>{item.value}</Text>
          <Text style={styles.textDate}> Task</Text>
        </View>
        <TouchableOpacity
          style={styles.iconContainer}
          onPress={() => editItem(item.key)}
        >
          <MaterialIcons name="edit" size={24} color="midnightblue" />
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.iconContainer}
          onPress={() => deleteItem(item.key)}
        >
          <MaterialIcons name="delete" size={24} color="midnightblue" />
        </TouchableOpacity>
      </TouchableOpacity>
    </View>
  );
}

const styles = {
  componentContainer: {
    flexDirection: "row",
    justifyContent: "center",
    height: "auto",
    width: "auto",
  },
  listContainer: {
    backgroundColor: "whitesmoke",
    height: "auto",
    width: 350,
    marginBottom: 30,
    borderRadius: 10,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  textItem: {
    color: "black",
    width: 260,
    height: "auto",
    fontSize: 20,
    marginTop: 10,
    marginRight: 20,
    fontFamily: "poppins-regular",
  },
  textDate: {
    color: "goldenrod",
    fontSize: 15,
    marginRight: 20,
    fontFamily: "poppins-regular",
    borderRadius: 10,
    width: 40,
  },
  iconContainer: {
    alignItems: "center",
    justifyContent: "center",
    marginRight: 0,
    marginTop: 15,
    height: 40,
    borderRadius: 10,
  },
  circleContainer: {
    alignItems: "center",
    justifyContent: "center",
    paddingLeft: 5,
  },
};
