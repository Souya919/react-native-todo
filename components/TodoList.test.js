import React from "react";
import renderer from "react-test-renderer";
import TodoList from "./TodoList";

// Mock the necessary dependencies
jest.mock("@expo/vector-icons", () => ({
  Entypo: "Entypo",
  MaterialIcons: "MaterialIcons",
}));

describe("<TodoList />", () => {
  it("renders correctly", () => {
    // Create a sample item object that matches the prop shape expected by TodoList
    const item = {
      key: "1",
      value: "Sample task",
    };

    // Define mock editItem and deleteItem functions
    const editItem = jest.fn();
    const deleteItem = jest.fn();

    // Render the TodoList component with the required props
    const tree = renderer
      .create(
        <TodoList item={item} editItem={editItem} deleteItem={deleteItem} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
