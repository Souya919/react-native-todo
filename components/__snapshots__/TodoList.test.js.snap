// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "center",
      "width": "auto",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "whitesmoke",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": "auto",
        "justifyContent": "space-between",
        "marginBottom": 30,
        "opacity": 1,
        "width": 350,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingLeft": 5,
        }
      }
    >
      <Entypo
        color="midnightblue"
        name="circle"
        size={20}
      />
    </View>
    <View>
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "poppins-regular",
            "fontSize": 20,
            "height": "auto",
            "marginRight": 20,
            "marginTop": 10,
            "width": 260,
          }
        }
      >
        Sample task
      </Text>
      <Text
        style={
          {
            "borderRadius": 10,
            "color": "goldenrod",
            "fontFamily": "poppins-regular",
            "fontSize": 15,
            "marginRight": 20,
            "width": 40,
          }
        }
      >
         Task
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 40,
          "justifyContent": "center",
          "marginRight": 0,
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <MaterialIcons
        color="midnightblue"
        name="edit"
        size={24}
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderRadius": 10,
          "height": 40,
          "justifyContent": "center",
          "marginRight": 0,
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <MaterialIcons
        color="midnightblue"
        name="delete"
        size={24}
      />
    </View>
  </View>
</View>
`;
